contracts/InterchainProposalSender.sol

```diff
@@ -44,154 +45,87 @@ contract InterchainProposalSender is IProposalSender {
         // revert if the sum of given fees are not equal to the msg.value
-        revertIfInvalidFee(fees);
+        revertIfInvalidFee(xCalls);
 
-        // revert if the length of given arrays are not equal
-        revertIfInvalidLength(
-            destinationChains,
-            destinationContracts,
-            fees,
-            targets,
-            values,
-            signatures,
-            data
-        );
-
-        for (uint i = 0; i < destinationChains.length; i++) {
-            _broadcastProposalToChain(
-                destinationChains[i],
-                destinationContracts[i],
-                fees[i],
-                targets[i],
-                values[i],
-                signatures[i],
-                data[i]
-            );
+        for (uint i = 0; i < xCalls.length; ) {
+            _sendProposal(xCalls[i]);
+            unchecked {
+                ++i;
+            }
         }
     }
 
     /**
      * @dev Broadcast the proposal to be executed at single destination chain.
      * @param destinationChain destination chain
      * @param destinationContract destination contract
-     * @param targets An array of contracts to call
-     * @param values An array of amounts of native tokens to send
-     * @param signatures An array of function signatures
-     * @param data An array of encoded function arguments
+     * @param calls An array of calls to be executed at the destination chain. Where each call contains the following:
+     * - target: target contract
+     * - value: amount of tokens to send
+     * - callData: encoded function arguments
      */
-    function broadcastProposalToChain(
+    function sendProposal(
         string memory destinationChain,
         string memory destinationContract,
-        address[] memory targets,
-        uint256[] memory values,
-        string[] memory signatures,
-        bytes[] memory data
-    ) external payable {
-        _broadcastProposalToChain(
-            destinationChain,
-            destinationContract,
-            msg.value,
-            targets,
-            values,
-            signatures,
-            data
+        InterchainCalls.Call[] calldata calls
+    ) external payable override {
+        _sendProposal(
+            InterchainCalls.InterchainCall(
+                destinationChain,
+                destinationContract,
+                msg.value,
+                calls
+            )
         );
     }
 
-    function _broadcastProposalToChain(
-        string memory destinationChain,
-        string memory destinationContract,
-        uint fee,
-        address[] memory targets,
-        uint256[] memory values,
-        string[] memory signatures,
-        bytes[] memory data
+    function _sendProposal(
+        InterchainCalls.InterchainCall memory xCall
     ) internal {
-        revertIfInvalidProposalArgs(targets, values, signatures, data);
-
-        if (fee == 0) {
+        if (xCall.gas == 0) {
```

Comment:
maybe the fee is being subsidized in which case the gas payment can be skipped? could be useful to allow that option